src/common/entryPoint.ts

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import {TelemetryHelper} from "../common/telemetryHelper";
+import {Telemetry} from "../common/telemetry";
+import {Log} from "../common/Log";
+import {OutputChannel} from "vscode";
+
+/* This class should we used for each entry point of the code, so we handle telemetry and error reporting properly */
+export class EntryPoint {
+    private outputChannel: OutputChannel;
+
+    constructor(outputChannel?: OutputChannel) {
+        this.outputChannel = outputChannel;
+    }
+
+    /* This method should wrap any async entry points to the code, so we handle telemetry and error reporting properly */
+    public runCode(taskName: string, errorDescription: string, codeToRun: () => Q.Promise<void> | void, errorsAreFatal: boolean): void {
```

Comment:
This is intended for code that is called from outside our application. For the time being none of those use-cases require us to return a value. If we find one in the future, we can change the typing then.